pysprint_tracks.py

```diff
@@ -999,31 +1006,32 @@ def blit_obstacles(self, race_started, overlay_blitted = False):
                     game_display.blit(poles_frames[self.poles_frame_indexes[2]],self.internal_pole_position)
 
         #Display spills
-        if self.display_oil_spill:
-            if self.oil_spill_position is None:
-                result = self.get_random_position(oil_spill_image.get_height(),oil_spill_image.get_width())
-                self.oil_spill_position = result[0]
-                self.on_bridge_oil_spill = result[1]
-            if race_started:
-                if not overlay_blitted or self.on_bridge_oil_spill:
-                    game_display.blit(oil_spill_image,self.oil_spill_position)
-
-        if self.display_water_spill:
-            if self.water_spill_position is None:
-                result = self.get_random_position(water_spill_image.get_height(),water_spill_image.get_width())
-                self.water_spill_position = result[0]
-                self.on_bridge_water_spill = result[1]
-            if race_started:
-                if not overlay_blitted or self.on_bridge_water_spill:
-                    game_display.blit(water_spill_image,self.water_spill_position)
-        if self.display_grease_spill:
-            if self.grease_spill_position is None:
-                result = self.get_random_position(grease_spill_image.get_height(),grease_spill_image.get_width())
-                self.grease_spill_position = result[0]
-                self.on_bridge_grease_spill = result[1]
-            if race_started:
-                if not overlay_blitted or self.on_bridge_grease_spill:
-                    game_display.blit(grease_spill_image,self.grease_spill_position)
+        for spill in self.oil_spills:
+
+            if not race_started:    
+                if spill.pos is None:
+                    pos, is_on_object = self.get_random_position(oil_spill_image.get_height(), oil_spill_image.get_width())
+                    spill.update(pos) if not is_on_object else spill.disable()
+            elif not overlay_blitted: #or self.on_bridge_oil_spill:
```

Comment:
@shazz Why did you comment the ` or self.on_bridge_oil_spill:`? The obstacles will not be blitted after the bridges are blitted. I couldn't test properly as poles and spills never appear on bridges in your PR. Cones are working fine on the  other hand. Maybe reuse the same logic?